x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/AsyncSearchTask.java

```diff
@@ -403,6 +405,7 @@ protected void onFetchFailure(int shardIndex, SearchShardTarget shardTarget, Exc
         protected void onListShards(List<SearchShard> shards, List<SearchShard> skipped, Clusters clusters, boolean fetchPhase) {
             // best effort to cancel expired tasks
             checkCancellation();
+            ccsMinimizeRoundtrips = (clusters.isCcsMinimizeRoundtrips() && clusters.getRemoteClusters() > 0);
```

Comment:
nit: do we need to set ccsMinimizeRoundtrips to true when there are no remote clusters? Maybe we could default  it to false instead?

Reply:
I think that's what I am doing?

Technically the Clusters object should know whether we are doing minimizeRoundtrips, so we should just be able to do `ccsMinimizeRoundtrips = clusters.isCcsMinimizeRoundtrips()` but I decided to add the extra protection of ensure there is at least one remote cluster.

Actually, maybe the best solution is to put an assert in the new `Clusters` constructor that takes the new arguments:

```
assert (ccsMinimizeRoundtrips ? remoteClusters > 0 : true) : "ccsMinimizeRoundtrips is true but remoteClusters is " + remoteClusters;
```

Then we can just do `ccsMinimizeRoundtrips = clusters.isCcsMinimizeRoundtrips()` here. Let me know if that sounds best to you.

Reply:
++

Reply:
this sounds like a good idea to me.

Reply:
Fixed in most recent commit.